packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.mdx

```diff
@@ -53,6 +53,27 @@ switch (value) {
 
 Since `value` is a non-union type it requires the switch case to have a default clause only with `requireDefaultForNonUnion` enabled.
 
+### `considerDefaultExhaustiveForUnions`
+
+Defaults to false. Whether a `default` clause will be considered as making a switch statement over a union type exhaustive.
+
+If set to true, `switch` statement over a union type includes a `default` case is considered exhaustive.
+`considerDefaultExhaustiveForUnions` enforces explicitly handling every constituent of the union type with their own explicit `case`.
+This can be useful if you want to make sure every value added to the union receives explicit handling, with the `default` case reserved for reporting an error.
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/pull/9954#discussion_r1801500418 applies to the other lines here too...

```suggestion
If set to true, a `switch` statement over a union type that includes a `default` case is considered exhaustive.
Otherwise, the rule enforces explicitly handling every constituent of the union type with their own explicit `case`.
Keeping this option disabled can be useful if you want to make sure every value added to the union receives explicit handling, with the `default` case reserved for reporting an error.
```